### PR TITLE
finished task4

### PR DESCRIPTION
**Pull request by Jay-R-Mac**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -44,8 +44,52 @@ describe("GET /api", () => {
...
+  it("responds with a 200 status message", () => {
+    return request(app).get("/api/articles/2").expect(200);
+  });
+  it("responds with an object which contains article information", () => {
+    return request(app)
+      .get("/api/articles/2")
+      .then(({ body }) =>
+        expect(body).toMatchObject({
+          article_id: expect.any(Number),
+          title: expect.any(String),
```
Comment (Jodiech) [inline __tests__/app.test.js:64]: Nice use of toMatchObject here 👍 make sure that you are asserting for the actual number of ID in the tests - you are expecting 2 here 😄

File: __tests__/app.test.js [hunk 2]
```diff
@@ -44,8 +44,52 @@ describe("GET /api", () => {
...
+          body: expect.any(String),
+          created_at: expect.any(String),
+          votes: expect.any(Number),
+          article_img_url: expect.any(String),
+        })
+      );
+  });
+  it("responds with a 404 when an article does not exist", () => {
+    return request(app).get("/api/articles/200000").expect(404);
+  });
```
Comment (Jodiech) [inline __tests__/app.test.js:73]: You can remove the first 404 test, as they are virtually the same 👍

File: __tests__/app.test.js [hunk 3]
```diff
@@ -44,8 +44,52 @@ describe("GET /api", () => {
...
+    return request(app)
+      .get("/api/articles/200000")
+      .expect(404)
+      .then(({ text }) => {
+        expect(JSON.parse(text)).toEqual({ message: "Article Not Found" });
+      });
+  });
+  it("responds with a 400 when an invalid id is passed", () => {
+    return request(app).get("/api/articles/hello").expect(400);
+  });
```
Comment (Jodiech) [inline __tests__/app.test.js]: You can remove this one too! 😄

File: model/model.js [hunk 4]
```diff
@@ -6,4 +6,20 @@ function getTopics() {
...
 }
 
-module.exports = { getTopics };
+function getArticleId(articleId) {
+  return db
+    .query("SELECT * FROM articles WHERE article_id =$1;", [articleId])
+    .then(({ rows }) => {
+      const foundArticle = rows[0];
+      if (!foundArticle) {
+        return Promise.reject({
```
Comment (Jodiech) [inline model/model.js:14]: Nice that you have handled this with a Promise.reject 👍

File: controller/controller.js [hunk 5]
```diff
@@ -11,4 +11,16 @@ const sendTopics = function (req, res, next) {
...
   res.status(200).send(endpoints);
 };
-module.exports = { sendTopics, sendEndpoints };
+
+const sendArticleId = function (req, res, next) {
+  const articleId = Number(req.params.article_id);
+  getArticleId(articleId)
+    .then((article) => {
+      res.status(200).send(article);
+    })
```
Comment (Jodiech) [inline controller/controller.js:21]: You could refactor this to `const {article_id} = req.params` and return the article restructured. Let me know if you need help with this 👍

**Review by Jodiech — CHANGES_REQUESTED** [review]
Hey 👋 nice work on this, please remove the two tests for 404 and 400, I have left comments for you to look at, give me a shout if you need any help 👍 Please resubmit when you're done!

File: controller/controller.js (hunk 5 above)
Comment (Jay-R-Mac) [reply controller/controller.js:21]: oops didn't see this will update this now

**Review by Jodiech — APPROVED** [review]

**Review by Jodiech — APPROVED** [review]
Happy to merge 👍

**Pull request merged**